### ISSUE TITLE
[Bug] The RPC call 'log_in" does not rise an error when the fingerprint is not in the keychain

### ISSUE THREAD

**Issue by maito78** [Chia-Network/chia-blockchain#18771]
### What happened?

When you log in with a random fingerprint `chia rpc wallet log_in '{"fingerprint": 1111111}'` you get:
```
{
  "fingerprint": 1111111,
  "success": true
}
```

### Version

2.4.4

### What platform are you using?

Linux

### What ui mode are you using?

CLI

**Comment by wjblanke** [comment]
This was done to support cases where the user deletes the key that is being used as the default wallet. the rpc will then pick another default if it can't find what it is looking for and return success. we could fix this better so the rpc is accurate and change the logic for default key selection but it may be pretty involved. will keep this issue open. is there some functionality that this causes issues for you?

**Comment by maito78** [comment]
Thank you for the answer. From my side is fine, I am using only fingerprints that exist, but I discovered it because I was debugging some code from the cli. I typed the fingerprint with one wrong digit, and I was confused because when I checked with `get_logged_in_fingerprint` I though I was in the right wallet (it was an existing fingerprint with one digit missing), but when printing the transactions I was getting another wallet. I think the transaction success message is fine, but the displayed fingerprint should be the one actually used and not the wrong one.
